nupic/algorithms/anomaly.py

```diff
@@ -130,15 +132,15 @@ def compute(self, activeColumns, predictedColumns,
     if self._mode == Anomaly.MODE_PURE:
       score = anomalyScore
     elif self._mode == Anomaly.MODE_LIKELIHOOD:
-      if inputValue is None:
-        raise ValueError("Selected anomaly mode 'Anomaly.MODE_LIKELIHOOD' "
-                 "requires 'inputValue' as parameter to compute() method. ")
-
+      assert inputValue is not None, ("Selected anomaly mode "
```

Comment:
Don't use asserts when you really want to raise an exception. Assert statements are discarded when you run code in "optimized" mode so they are only appropriate for invariance checks that would be considered bugs if they were ever true.